src/main/java/com/github/dockerjava/jaxrs/filter/ResponseStatusExceptionFilter.java

```diff
@@ -79,13 +79,18 @@ private String getBodyAsMessage(ClientResponseContext responseContext) {
                 String message = IOUtils.toString(entityStream, charset);
 
                 if (MediaType.APPLICATION_JSON_TYPE.equals(mediaType)) {
-                    ObjectMapper mapper = new ObjectMapper();
-                    JsonNode node = mapper.readTree(entityStream).get("message");
-                    if (node != null) {
-                        message = node.textValue();
+                    try {
+                        JsonNode node = new ObjectMapper().readTree(message);
+                        if (node != null) {
+                            JsonNode messageNode = node.get("message");
+                            if (messageNode != null && messageNode.isTextual()) {
+                                message = messageNode.textValue();
+                            }
+                        }
+                    } catch (IOException ignored) {
+                        //ignore parsing errors and return the message as is
```

Comment:
please log to error logger, that may help to identify issue and fix in future

Reply:
Done, I also logged the stack trace. let me know if you prefer to log only the error message